opnsynid_payment_instruction_aeroo_report/reports/account_payment_instruction_helper.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# © 2017 OpenSynergy Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api
+from openerp.tools import drop_view_if_exists
+from datetime import datetime
+
+
+class AccountPaymentInstructionHelper(models.Model):
+
+    _name = 'account.payment_instruction_helper'
+    _description = 'Payment Instruction Helper'
+    _auto = False
+
+    @api.multi
+    def _get_date(self):
+        for record in self:
+            payment_line = record.line_id
+
+            order = payment_line.order_id
+            if order.date_prefered == 'fixed':
+                record.date = payment_line.date
+            elif order.date_prefered == 'now':
+                record.date = order.date_created
+            else:
+                if payment_line.ml_maturity_date:
+                    record.date = payment_line.ml_maturity_date
+                elif payment_line.ml_date_created:
+                    record.date = payment_line.ml_date_created
+                else:
+                    record.date = order.date_created
+
+    @api.multi
+    def _get_date_test(self):
+        for record in self:
+            payment_line = record.line_id
+
+            order = payment_line.order_id
+            if order.date_prefered == 'fixed':
+                record.date_test = payment_line.date
+            elif order.date_prefered == 'now':
+                record.date_test = order.date_created
+            else:
+                if payment_line.ml_maturity_date:
+                    record.date_test = payment_line.ml_maturity_date
+                elif payment_line.ml_date_created:
+                    record.date_test = payment_line.ml_date_created
+                else:
+                    record.date_test = order.date_created
+
+    line_id = fields.Many2one(
+        string='Payment Lines',
+        comodel_name='payment.line'
+    )
+
+    date = fields.Date(
+        string="Date",
+        store=True
+    )
+
+    def init(self, cr):
+        drop_view_if_exists(cr, 'account_payment_instruction_helper')
+        strSQL = """
+            CREATE OR REPLACE VIEW account_payment_instruction_helper AS (
+                SELECT
+                    row_number() OVER() as id,
+                    A.id AS line_id,
+                    CASE 
+                        WHEN B.date_prefered = 'fixed'
+                            THEN B.date_scheduled
+                        WHEN B.date_prefered = 'now'
+                            THEN B.date_created
+                        ELSE
+                            CASE
+                            WHEN C.date_maturity NOTNULL 
+                                THEN C.date_maturity
+                            WHEN C.date NOTNULL
```

Comment:
Gunakan IS NOT NULL, NOTNULL tidak standard menurut https://www.postgresql.org/docs/9.1/static/functions-comparison.html